### ISSUE TITLE
Using this code with own data 

### ISSUE THREAD

**Issue by antithing** [dylan-campbell/bpnpnet#1]
Hi! And thanks for making this code available.

I have a data set that consists of:

A set of 3d points (xyz values);
A set of 2d points (xy pixel values)
A camera matrix (fx,fy,cx,cy, 5 distortion params) 

Is it possible to run your code with this data? 

Also, related question:

Where can I pass in my own data?

Thanks!

**Comment by dylan-campbell** [comment]
Hi! Yes, the network is designed for pure geometric inputs (3D and 2D points, without visual features or RGB values), so your data should be well suited.

Your 2D points should be K-normalised dehomog( K^{-1} homog(p) ), where homog converts to homogeneous coordinates and dehomog reverses that.

And you'll need to write your own Dataset class (see utilities/dataset_utilities.py), where __getitem__ returns a tuple of the form:
2d_pointcloud, 3d_pointcloud, true_rotation_matrix, true_translation, true_correspondences, num_points_2d, num_points_3d

**Comment by antithing** [comment]
Hi Dylan, thank you for getting back to me. 

Can you please explain this for me?

`Your 2D points should be K-normalised dehomog( K^{-1} homog(p) ), where homog converts to homogeneous coordinates and dehomog reverses that.`


What does that look like in code?  Thanks again

**Comment by dylan-campbell** [comment]
It just means that you append a 1 to your 2D coordinates, pre-multiply by inverse K (camera matrix), and then remove the final coordinate (which will be a 1).

Something like
p = torch.matmul(torch.inverse(K),  torch.nn.functional.pad(p, (0, 1), "constant", 1.0))
p = p[:, :2]
